### PR TITLE
FIX document references, remove some documentation files not needed anymore and more minor fixes

### PR DESCRIPTION
**Pull request by fgalan**
This is part of a bigger series https://github.com/telefonicaid/fiware-orion/pull/4141

Commit [a47e8b5](https://github.com/telefonicaid/fiware-orion/pull/4196/commits/a47e8b50606c4d1821c455e6ba5925a6e3ba10e8) covers:

* Look in the whole documentation (and code, and test, etc.) for references to deleted md files and fix them, change the links to the proper section in the new orion-api.md document
* No direct reference to NGSIv2 spec should remain in the documentation (the reference in the Orion API preface should suffice). Existing references to NGSIv2 spec should be replaced to references to Orion API (hint: text search for "github.io")
* Remove structured_attribute_valued.md (already covered by "JSON Attribute Representation" section)
* Remove update_action_types.md (already covered by "POST /v2/op/update" section)
* Remove geolocation.md (the documenta is basically just a pointer, with no content)
* Move GeoJson limitations from "Geospatial properties of entities" to "GeoJSON" specific subsection

Commit [7b37638](https://github.com/telefonicaid/fiware-orion/pull/4196/commits/7b37638e83bb31bf59ad697f0fb1ddcd9055575a) covers

* Remove filtering.md (the document is basically just a set of pointers, with no content)

Commit [585590a](https://github.com/telefonicaid/fiware-orion/pull/4196/commits/585590a70f90a67b21bfbcf0f8f76b1d76726a17) covers:

* Integrate pagiantion.md into new section "Pagination"

Commit [1576b53](https://github.com/telefonicaid/fiware-orion/pull/4196/commits/1576b53f8368c10e4e35fa753b74f0b89b7798d0) covers:

*  Cite regex.md in the proper places (eg. when idPattern is mentioned, etc.)

### CODE REVIEW

**Comment by fgalan** [comment]
Script used to check for references:

```
for doc in forbidden_characters.md metadata.md multitenancy.md ngsiv2_implementation_notes.md oneshot_subscription.md service_path.md subscriptions_alttype.md transient_entities.md structured_attribute_valued.md update_action_types.md geolocation.md
do
  echo "$doc:"
  grep $doc -R fiware-orion/* | grep -v 'manuals.jp'
done

echo ngsiv2 spec:
grep github.io -R fiware-orion/* | grep v2 | grep -v 'manuals.jp' | grep -v apiary
```

Executing it at [a47e8b5](https://github.com/telefonicaid/fiware-orion/pull/4196/commits/a47e8b50606c4d1821c455e6ba5925a6e3ba10e8) we get:

```
forbidden_characters.md:
metadata.md:
fiware-orion/doc/manuals/admin/upgrading_crossing_1-5-0.md:This procedure has to be done *only* if your DB contains attributes with [ID metadata](https://github.com/telefonicaid/fiware-orion/blob/1.5.0/doc/manuals/user/metadata.md#metadata-id-for-attributes).
multitenancy.md:
ngsiv2_implementation_notes.md:
oneshot_subscription.md:
service_path.md:
subscriptions_alttype.md:
transient_entities.md:
structured_attribute_valued.md:
update_action_types.md:
geolocation.md:
ngsiv2 spec:
fiware-orion/doc/manuals/orion-api.md:This is the release 2.1 of the [NGSIv2 specification](http://telefonicaid.github.io/fiware-orion/api/v2/stable),
fiware-orion/doc/manuals/orion-api.md:original NGSIv2 specification](http://telefonicaid.github.io/fiware-orion/api/v2/stable/). These differences come after years of experience with NGSIv2, in two senses:
```

Which seems to be ok

File: README.md [hunk 1]
```diff
@@ -177,22 +177,18 @@ Reference sections below in order to know more details about the API
...
 ## Reference Documentation
 
 API Reference Documentation:
 
--   FIWARE NGSI v2
-    [(en)](http://telefonicaid.github.io/fiware-orion/api/v2/stable)
-    [(jp)](https://open-apis.letsfiware.jp/fiware-orion/api/v2/stable/) (Apiary)
+-   Orion API FIWARE NGSI v2
+    [(en)](doc/manuals/orion-api.md)
+    [(jp)](doc/manuals.jp/orion-api.md) (Markdown)
```
Comment (fgalan) [inline README.md:191]: @fisuda jus to check... will this be the path to the orion-api.md Japanese version when PR https://github.com/telefonicaid/fiware-orion/pull/4162 gets merged?

Comment (fisuda) [reply README.md:191]: Yes. I put orion-api.md in doc/manuals.jp.

**Review by mapedraza — APPROVED** [review]
LGTM

**Pull request merged**